scalafix-core/shared/src/main/scala/scalafix/internal/rule/ExplicitResultTypes.scala

```diff
@@ -71,20 +76,61 @@ case class ExplicitResultTypes(
     case _: Defn.Var => MemberKind.Var
   }
 
+  @tailrec private def isOwner(owner: Symbol, symbol: Symbol): Boolean =
+    if (symbol == owner) true
+    else {
+      symbol match {
+        case Symbol.None => false
+        case Symbol.Global(qual, _) => isOwner(owner, qual)
+        case _ => false
+      }
+    }
+
   override def fix(ctx: RuleCtx): Patch = {
+    val table = index.asInstanceOf[EagerInMemorySemanticdbIndex]
```

Comment:
It's due to the `SemanticdbIndex` trait having only one expected implementation. The trait exists to separate the public API from the internal API, nobody is expected to implement a custom `SemanticdbIndex`. I opened https://github.com/scalacenter/scalafix/issues/704 to track this, I'm afraid we'll have to fix this before v0.6.0 because the memory pressure from eagerly loading a full classpath is  high